### ISSUE TITLE
2FA Sign in stopped working

### ISSUE THREAD

**Issue by ptirth** [Jman012/Wasserflug-tvOS#19]
So I was logged out yesterday of the app and haven’t been able to log back in.

Getting 2FA error.
![image](https://user-images.githubusercontent.com/32465104/235009881-9511222a-4386-4d1e-99d2-73e8366dda7c.jpg)

PS: I’m in beta tvOS version, not sure if that affects it.

**Comment by Jman012** [comment]
Hi @ptirth. This error looks like Floatplane is rejecting your 2FA code. I would see if you are able to login to https://floatplane.com with the same username/email and password and 2FA code. If you are, then you should be able to login to Wasserflug as well with the same credentials.

**Comment by ptirth** [comment]
I just retried it in incognito mode, and it’s working fine. Only having issues on the Apple TV app.

Now I’m curious.

**Comment by Jman012** [comment]
Hmm how very strange.

There shouldn't be an issue with main vs beta versions. The login system hasn't been touched much inbetween the two versions. And others can login just fine with their accounts, including me (tested just now). 

First step of narrowing down the problem would be to ensure what you're inputting on the website and in the app are the same. I would initially think of similar looking characters, or an accidental space somewhere. But that's more likely to occur for the username/password. You got through on those steps though because it's asking for the 2FA code and wouldn't get there if the password was incorrect. So, I guess make sure the 2FA is purely just numerical digits; no spaces or hyphens anywhere in it. If this is the problem, I can normalize the user input before sending it to the server to help prevent this in the future.

Second step would be determining why this is happening to your/your account and not others. There really shouldn't be anything on Floatplane's end that would cause this, I would think. And the logging in system on Wasserflug isn't so complicated that this could be fixed by deleting and re-installing the app on your TV.

I would ask for retrieving logs from your Apple TV, but it requires not only a Mac laptop/desktop, but also Xcode which is asking a lot for some users. Or for me to try it myself, but it would require giving me your login info which is not recommended. 

See if a reinstall helps, try more carefully inputting the code without spaces or other extraneous characters, see if you can retrieve logs (https://developer.apple.com/documentation/xcode/acquiring-crash-reports-and-diagnostic-logs), and pray to the tech gods themselves? If nothing else works, feel free to message me on discord Jamamp#2672 and we can maybe hop on a private call to see what's happening in a debugging session.

**Comment by ptirth** [comment]
Okay, so I tried both, reinstalling the app and entering 2FA manually. And it worked. Not sure which one though :) I was using voice command from the remote to enter 2FA before.

PS: Something else I have noticed maybe recently? The selected video/post overlaps it's surrounding posts when it zoom in? I actually forgot if it was present before or not. I can certainly open new issue if necessary, just wanted to get it in your attention.

![IMG_1816](https://user-images.githubusercontent.com/32465104/235055118-fded5503-71a1-43bf-b621-d4cb6d15a0aa.jpeg)

**Comment by Jman012** [comment]
Huzzah! I'll make a note to try to help normalize the text from voice inputs. There might be some invisible character going through to Floatplane.

For the tiles, tvOS 16.4 messed with SwiftUI buttons (again, after 16.0 also messed with things). I made a fix for it here with some more details: https://github.com/Jman012/Wasserflug-tvOS/commit/68a1675744b7b342e73fbd38a8588e6225a1c7ba

I got that pushed out in version 1.3.0(11). You might still be on 1.3.0(10) :). Use the Test Flight app to get it updated to the latest and that should be all squared away!
